packages/workspace/src/browser/workspace-duplicate-handler.ts

```diff
@@ -0,0 +1,46 @@
+
+/********************************************************************************
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { UriCommandHandler } from '@theia/core/lib/common/uri-command-handler';
+import URI from '@theia/core/lib/common/uri';
+import { injectable, inject } from 'inversify';
+import { FileSystem } from '@theia/filesystem/lib/common/filesystem';
+
+@injectable()
+export class WorkspaceDuplicateHandler implements UriCommandHandler<URI[]> {
+
+    @inject(FileSystem)
+    protected readonly fileSystem: FileSystem;
+
+    async execute(uris: URI[]): Promise<void> {
+        const distinctUris = URI.getDistinctParents(uris);
+        await Promise.all(distinctUris.map(uri => this.duplicate(uri)));
+    }
+
+    duplicate(uri: URI) {
+        try {
+            let name = uri.displayName;
+            const nameParts = name.split('.');
+            nameParts[0] = nameParts[0] + ' copy';
```

Comment:
What is there is already a file with such name? Please look at how other handlers deal with it, there should be a function to generate a unique name.